README.md

```diff
@@ -1 +1,308 @@
+
+This repository contains code and results for paper: Measure this, not that: Optimizing the cost and model-based information content of measurements 
+
+## Installation instructions 
+
+The following instructions assume you have anaconda installed. We suggest create an environment with the following commands to run code: 
+
+### Step 1: create a new environment 
+- create new environment, called for e.g. `measurement_optimization`, with `conda` with `Python` version 3.8
+
+`conda create --name measurement_optimization python=3.8`
+
+`conda activate measurement_optimization`
+   
+### Step 2: install `IDAES-PSE`
+- this step provides `Ipopt` solver, but this solver is not necessary for reproducing paper results
+- we did not test if the code can run without this step
+
+`pip install idaes-pse` 
+
+`idaes get-extensions`
+   
+### Step 3: install `Pyomo` from specified branches
+- install from the following branch for a generalization version of Mindtpy:
+
+`pip install git+https://github.com/ZedongPeng/pyomo.git@add_mindtpy_callback`
+
+   
+### Step 4: install `GurobiPy`
+- this is needed only for solving mixed-integer problems
+
+  `conda install -c gurobi gurobi`
+   
+### Step 5: install `CyIpopt`
+- this is needed only for D-optimality problems with grey-box modules
+
+   `conda install -c conda-forge cyipopt`
+
+### Software versions we use for the results 
+
+`Python`: 3.8
+
+`IDAES-PSE`: 2.2.0
+
+`Pyomo`: 6.7.0 dev 0
+
+`GurobiPy`: 10.0.3
+
+`CyIpopt`: 1.3.0
+
+## Code 
+
+- `measure_optimize.py`: Measurement optimization optimization framework
+
+- `greybox_generalize.py`: Grey-box generalization 
+
+- `kinetics_MO.py`: Kinetics case study
+
+- `rotary_bed_MO.py`: Rotary bed case study
+
+- `Draw_figure.ipynb`: Generates all results figures in the manuscript
+
+
+## Example to run code and reproduce figures for case studies
+
+Setup the scripts to reproduce result files and figures from the paper: 
+
+### Kinetics case study 
+
+- Step 1: run `kinetics_MO.py`
+- Step 2: with `mip_option` and `objective`, choose to run the A-optimality or D-optimality, mixed-integer or relaxed problem 
+- Step 3: with `budget_ranges`, set up the budget ranges as you want to try. The first three budgets will run as a test.
+  Suggested ranges: 
+
+  In our results, we use the budget range [1000, 5000] with a 100 discretization,
+  i.e. [1000, 1100, ..., 5000], for relaxed problems
+
+  we use the budget range [1000, 5000] with a 400 discretization,
+  i.e. [1000, 1400, 1800, ..., 5000] for mixed-integer problems.
+
+  You can choose the same budget ranges, or only a few points of budgets, to run the script.
+
+- Step 4: with `initializer_option` and `curr_results`, select initial solutions to initialize the problem, and provide file paths for these solutions
+
+  You can choose from: A- and D-optimality, with mixed-integer or continuous options. 
```

Comment:
What was used in the paper? Please elaborate here.

Reply:
Used all four. I added a sentence about which equations are A- and D- in the paper that people should refer to.